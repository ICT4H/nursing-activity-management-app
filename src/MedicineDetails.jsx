import React from 'react';
class MedicineDetails extends React.Component {
    render(){
        const medicine=this.props.medicine;
        return (<div className={"medicineDetails"}>
      <p>{medicine.medicineName}</p>
      <p>{medicine.dose} {medicine.unit}</p>
    </div>)
    }
}
export default MedicineDetails;