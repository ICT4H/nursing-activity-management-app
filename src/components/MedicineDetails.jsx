import React from 'react';
class MedicineDetails extends React.Component {
    render(){
        const medicine=this.props.medicine;
        return (<div className={"medicineDetails"} onClick={this.props.onClick}>
      <p>{medicine.medicineName}</p>
      <p>{medicine.dose} {medicine.unit}</p>
    </div>)
    }
}
export default MedicineDetails;