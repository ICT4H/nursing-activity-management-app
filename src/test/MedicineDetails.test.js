import React from 'react';
import renderer from 'react-test-renderer';
import MedicineDetails from "../MedicineDetails";

test('Given medicine object with medicineName,dose and unit gives all in "p" tag and wrapped around with "div" tag', () => {
  let medicine={medicineName:"Paracetmol", dose:2,unit:"table"};
  const component = renderer.create(
      <MedicineDetails medicine={medicine}/>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});