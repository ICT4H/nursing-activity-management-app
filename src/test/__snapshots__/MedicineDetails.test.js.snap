// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given medicine object with medicineName,dose and unit gives all in "p" tag and wrapped around with "div" tag 1`] = `
<div
  className="medicineDetails"
  onClick={undefined}
>
  <p>
    Paracetmol
  </p>
  <p>
    2	table
  </p>
</div>
`;
